### PR TITLE
[Cart] Documentation of Cart component

### PR DESCRIPTION
**Pull request by Arminek**
| Q                    | A
| -------------------| ---
| Doc fix?          | [yes]
| New docs?     | [no] 
| Fixed tickets   | [no]

### CODE REVIEW

File: components/Cart/index.rst [hunk 1]
```diff
@@ -2,14 +2,12 @@ Cart
 ====
 
 Common models, services and interface to handle a shopping cart in PHP e-commerce application.
+It is strict related to Order model.
```
Comment (TheMadeleine) [inline components/Cart/index.rst]: `It is strictly related to the **Order** model.`

File: components/Cart/interfaces.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    For more detailed information go to `Sylius API's CartItemInterface`_.
+
+.. _Sylius API's CartItemInterface: http://api.sylius.org/Sylius/Component/Cart/Model/CartItemInterface.html
+
+Service Interfaces
+------------------
+
+.. _component_cart_provider_cart-provider-interface:
+
+Cart provider
```
Comment (TheMadeleine) [inline components/Cart/interfaces.rst]: `CartProviderInterface` - I suppose we should use full names in the headlines for all interfaces below.

File: components/Cart/interfaces.rst [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+~~~~~~~~~~~~~
+
+A cart provider retrieves existing cart or create new one based on storage. To characterize an object which is a **Provider**,
+it needs to implement the ``CartProviderInterface``.
```
Comment (TheMadeleine) [inline components/Cart/interfaces.rst]: I'm not really satisified with such descriptions of interfaces. Thay should be a bit more... human :) 
 Maybe : 
`This interface gives provides a possibility to retrieve an existing cart or create a new one basing on the storage.` ?

File: components/Cart/interfaces.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+A cart resolver returns cart item that needs to be added based on given data. To characterize an object which is a **Resolver**,
+it needs to implement the ``ItemResolverInterface``.
+
+.. note::
+    For more detailed information go to `Sylius Api ItemResolverInterface`_.
+
+.. _Sylius Api ItemResolverInterface: http://api.sylius.org/Sylius/Component/Cart/Resolver/ItemResolverInterface.html
+
+.. caution::
+    This method throw `ItemResolvingException`_ if an error occurs.
```
Comment (TheMadeleine) [inline components/Cart/interfaces.rst]: ``It throws an `ItemResolvingException`_. ``

File: components/Cart/interfaces.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+.. _ItemResolvingException: http://api.sylius.org/Sylius/Component/Cart/Resolver/ItemResolvingException.html
+
+.. _component_cart_repository_cart-repository-interface:
+
+CartRepositoryInterface
+~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to decouple from storage that provides expired carts, you should create repository class which implements this interface.
```
Comment (TheMadeleine) [inline components/Cart/interfaces.rst]: :+1:

File: components/Cart/interfaces.rst [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+.. note::
+    This interface extends the :ref:`component_order_repository_order-repository-interface` |br|
+    For more detailed information go to `Sylius API's CartRepositoryInterface`_.
+
+.. _Sylius API's CartRepositoryInterface: http://api.sylius.org/Sylius/Component/Cart/Repository/CartRepositoryInterface.html
+
+CartContextInterface
+~~~~~~~~~~~~~~~~~~~~
+
+This interface is implemented by the service responsible for setting and retrieving current cart identifier based on storage.
```
Comment (TheMadeleine) [inline components/Cart/interfaces.rst]: `by the services [...]`

File: components/Cart/models.rst [hunk 7]
```diff
@@ -14,13 +16,15 @@ The cart is represented by **Cart** instance. It inherits all the properties fro
...
-    This model implements ``CartInterface``
+    This model implements :ref:`component_cart_model_cart-interface`
+
+.. _component_cart_model_cart-item:
 
 CartItem
 --------
 
-The items of the cart are represented by **CartItem** instance and it has not property but it inherits all of them from ``Sylius\Component\Order\Model\OrderItem``
+The items of the cart are represented by **CartItem** instance and it has no property but it inherits all of them from :ref:`component_order_model_order-item`
```
Comment (TheMadeleine) [inline components/Cart/models.rst]: ``The items in the cart are represented by the **CartItem** instances. **CartItem** has no properties but it inherits from :ref:`component_order_model_order-item`. ``

File: components/Cart/models.rst [hunk 8]
```diff
@@ -14,13 +16,15 @@ The cart is represented by **Cart** instance. It inherits all the properties fro
...
 
 .. note::
 
-    This model implements ``CartItemInterface``
\ No newline at end of file
+    This model implements :ref:`component_cart_model_cart-item-interface`
```
Comment (TheMadeleine) [inline components/Cart/models.rst]: implements **the** :ref:(...)

File: components/Cart/interfaces.rst [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+.. _Sylius API's CartRepositoryInterface: http://api.sylius.org/Sylius/Component/Cart/Repository/CartRepositoryInterface.html
+
+CartContextInterface
+~~~~~~~~~~~~~~~~~~~~
+
+This interface is implemented by the services responsible for setting and retrieving current cart identifier based on storage.
+To characterize an object which is a **CartContext** it needs to implement the ``CartContextInterface``
+
+.. note::
+    For more detailed information go to `Sylius API CartContextInterface`_.
```
Comment (tuka217) [inline components/Cart/interfaces.rst]: You use many variant of text "sylius api..". Could you choose one and use in all files?

Comment (pjedrzejewski) [inline components/Cart/interfaces.rst]: As long as it makes sense, I am fine with different text here. ;)

**Comment by pjedrzejewski** [comment]
Thanks Arek!

**Pull request merged**